### PR TITLE
Fixes misalignment on template list and advanced search Key dropdown bug

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses #10642 and #10611.  I found some lists that did not have anything in the Advanced Search ---> Key drop down.  This should address all of those issues also.
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION
Screen Shot 2021-08-27 at 3 58 25 PM
Screen Shot 2021-08-27 at 3 58 12 PM

### CODE REVIEW

File: awx/ui/src/screens/Host/HostList/HostList.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ function HostList() {
         actions: results[1].data.actions,
         relatedSearchableKeys: (
           results[1]?.data?.related_search_fields || []
-        ).map((val) => val.slice(0, -8)),
+        ).map((val) => (val.endsWith('search') ? val.slice(0, -8) : val)),
```
Comment (nixocio) [inline awx/ui/src/screens/Host/HostList/HostList.js:71]: I liked this solution. Very readable. Good job @AlexSCorey.

File: awx/ui/src/screens/Inventory/InventorySources/InventorySourceList.js [hunk 2]
```diff
@@ -50,12 +57,20 @@ function InventorySourceList() {
         sourceCount: results[0].data.count,
         sourceChoices: results[1].data.actions.GET.source.choices,
         sourceChoicesOptions: results[1].data.actions,
+        searchableKeys: Object.keys(results[1].data.actions?.GET || {}).filter(
+          (key) => results[1].data.actions?.GET[key].filterable
+        ),
+        relatedSearchableKeys: (
+          results[1]?.data?.related_search_fields || []
+        ).map((val) => val.slice(0, -8)),
```
Comment (nixocio) [inline awx/ui/src/screens/Inventory/InventorySources/InventorySourceList.js:65]: Do we need to verify `__search` here?

Comment (AlexSCorey) [reply awx/ui/src/screens/Inventory/InventorySources/InventorySourceList.js:65]: No, not here.  It seems to be that only the `ansible_facts` related search key does not end in `__search`, which I could argue really is an API issue.  If/when it's fixed the [this line](https://github.com/ansible/awx/pull/10959/files#diff-18074f2fc3330129129451ed5e744021a2339f8c15b2e0afc624871b489518cbR71) won't be needed.  AFAIK `ansible_facts` only shows up in lists that have to do with hosts.

**Review by nixocio — APPROVED** [review]

**Review by akus062381 — APPROVED** [review]

**Comment by akus062381** [comment]
good as far as QE!

**Pull request merged**